lib/Doctrine/DBAL/Connections/MasterSlaveConnection.php

```diff
@@ -352,7 +356,11 @@ public function query()
             $logger->startQuery($args[0]);
         }
 
-        $statement = $this->_conn->query(...$args);
+        try {
+            $statement = $this->_conn->query(...$args);
+        } catch (Throwable $e) {
```

Comment:
Same as above: we upcast these, so if we need to further wrap it there's something else going on

Reply:
I follow the same idea from the regular Connection class https://github.com/doctrine/dbal/blob/master/lib/Doctrine/DBAL/Connection.php#L1011-L1015